src/interpolation/ArborX_InterpMovingLeastSquares.hpp

```diff
@@ -0,0 +1,274 @@
+                  "Source points elements must be points");
+    static constexpr int dimension = GeometryTraits::dimension_v<src_point>;
+
+    // TargetPoints is an access trait of points
+    ArborX::Details::check_valid_access_traits(PrimitivesTag{}, target_points);
+    using tgt_acc = AccessTraits<TargetPoints, PrimitivesTag>;
+    static_assert(KokkosExt::is_accessible_from<typename tgt_acc::memory_space,
+                                                ExecutionSpace>::value,
+                  "Target points must be accessible from the execution space");
+    using tgt_point =
+        typename ArborX::Details::AccessTraitsHelper<tgt_acc>::type;
+    GeometryTraits::check_valid_geometry_traits(tgt_point{});
+    static_assert(GeometryTraits::is_point<tgt_point>::value,
+                  "Target points elements must be points");
+    static_assert(dimension == GeometryTraits::dimension_v<tgt_point>,
+                  "Target and source points must have the same dimension");
+
+    int num_neighbors_val =
+        (!num_neighbors)
+            ? Details::polynomialBasisSize<dimension, PolynomialDegree::value>()
+            : *num_neighbors;
+
+    int const num_targets = tgt_acc::size(target_points);
+    _source_size = source_points.extent(0);
+    // There must be enough source points
+    KOKKOS_ASSERT(0 < num_neighbors_val && num_neighbors_val <= _source_size);
+
+    // Organize the source points as a tree
+    using src_coord = typename GeometryTraits::coordinate_type<src_point>::type;
+    using box = ExperimentalHyperGeometry::Box<dimension, src_coord>;
+    using bvh = BasicBoundingVolumeHierarchy<
+        MemorySpace, ArborX::Details::PairIndexVolume<box>,
+        ArborX::Details::DefaultIndexableGetter, box>;
+    bvh source_tree(
+        space, ArborX::Details::LegacyValues<SourcePoints, box>{source_points});
+
+    // Create the predicates
+    Details::MLSTargetPointsPredicateWrapper<TargetPoints> predicates{
+        target_points, num_neighbors_val};
+
+    // Query the source
+    Kokkos::View<int *, MemorySpace> indices(
+        "ArborX::MovingLeastSquares::indices", 0);
+    Kokkos::View<int *, MemorySpace> offsets(
+        "ArborX::MovingLeastSquares::offsets", 0);
+    source_tree.query(space, predicates,
+                      ArborX::Details::LegacyDefaultCallback{}, indices,
+                      offsets);
+
+    // Fill in the value indices object so values can be transferred from a 1D
+    // source data to a properly distributed 2D array for each target.
+    auto const source_view = fillValuesIndicesAndGetSourceView(
+        space, indices, offsets, num_targets, num_neighbors_val, source_points);
+
+    // Compute the Moving Least Squares
+    _coeffs = Kokkos::View<FloatingCalculationType **, MemorySpace>(
+        "ArborX::MovingLeastSquares::coefficients", 0, 0);
+    Details::movingLeastSquaresCoefficients<CRBF, PolynomialDegree>(
+        space, source_view, target_points, _coeffs);
```

Comment:
Is there a good reason not to return the coefficients?

Reply:
Like previously, I do not know if it is better to return the coefficients or to give an uninitialized array to it. Querying a tree requires to give out empty views for example.